plugins.go

```diff
@@ -106,8 +100,12 @@ func GetBackendWithParams(ctx context.Context, typeName string, params InitParam
 	if !exists {
 		return nil, fmt.Errorf("storage.type %q not found or registered", typeName)
 	}
-	if params.Logger.GetSink() == nil {
-		params.Logger = logr.Discard()
+	p := InitParams{OptionMap: options}
+	for _, param := range params {
+		param(&p)
+	}
+	if p.Logger.GetSink() == nil {
```

Comment:
`logr.Logger` is an interface, and this condition will trigger a runtime error if `p.Logger` is `nil`

Reply:
Logr v1 introduced a breaking change: logr.Logger is now a struct instead of an interface. See https://github.com/go-logr/logr/pull/42

See https://github.com/go-logr/logr/issues/152 for my suggestion for a cleaner way to check for a zero value.

Reply:
I didn't catch that change, LGTM then!